src/olympia/shelves/models.py

```diff
@@ -27,3 +27,15 @@ class Meta:
 
     def __str__(self):
         return self.title
+
+
+class ShelfManagement(ModelBase):
+    shelf = models.OneToOneField(Shelf, on_delete=models.CASCADE)
+    enabled = models.BooleanField(default=False)
+    position = models.PositiveIntegerField(blank=True, null=True)
```

Comment:
when is it necessary to have `position` is `None`?   Wouldn't you just delete the `ShelfManagement` object?

Reply:
Possibly. With the enabled field, the user has the option of keeping shelf/shelves without enabling them on the homepage. My concern was to catch the case where the admin is reordering enabled shelves and just forgets to set the position before saving.

Reply:
it doesn't default to `0`? (or some other non-None number)

Reply:
No, default appears to be None. Should I set it to 0?

Reply:
yes, i think so (it might happen automatically if `null=True` wasn't specified)

Reply:
It didn't happen automatically, but `def clean` in models would no longer be needed if default is set to `0`, correct?

Reply:
correct.  If the field didn't allow `None` as a value then it would trigger the field's validation instead.